ros_packages/voice_assistant/voice_assistant/chat.py

```diff
@@ -148,11 +195,12 @@ async def chat(self, goal_handle: ServerGoalHandle):
 
         try:
             # receive assistant-response in form of an iterable of tokens from the public-api
+            # message_history[-20] for limiting the sice of the history
             with self.public_voice_client_lock:
                 tokens = public_voice_client.chat_completion(
                     text=content,
                     description=description,
-                    message_history=message_history,
+                    message_history=message_history[-20:],
```

Comment:
If we don't use the whole `message_history`, ` voice_assistant_client.get_all_chat_messages` should be modified to have an optional `k` parameter (or similar) that allows retrieval of the latest `k` messages instead of querying and retrieving all of them.

Reply:
replaced with var

Reply:
No, I meant that `voice_assistant_client.get_all_chat_messages(chat_id)` should be changed into something like `voice_assistant_client.get_chat_messages(chat_id, k: int)` to only retrieve the specified amount (e.g. 20) from the DB directly.

Reply:
This should work from now on